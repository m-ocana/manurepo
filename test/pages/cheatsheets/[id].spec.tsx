import React from "react";
import { render, act } from "@testing-library/react";
import Cheatsheet from "../../../pages/cheatsheets/[id]";
import Prism from "prismjs";

jest.mock("prismjs", () => ({
  highlightAll: jest.fn(),
}));

describe("Cheatsheet page", () => {
  beforeEach(jest.clearAllMocks);

  it("renders a Cheatsheet page", () => {
    const rendered = render(
      <Cheatsheet name="Dummy Sheet" contentHtml={<div>Dummy Content</div>} />
    );
    expect(rendered.asFragment()).toMatchSnapshot();
  });

  it("loads Prism when the component mounts", () => {
    act(() => {
      render(
        <Cheatsheet name="Dummy Sheet" contentHtml={<div>Dummy Content</div>} />
      );
    });
    expect(Prism.highlightAll).toHaveBeenCalledTimes(1);
  });
});
