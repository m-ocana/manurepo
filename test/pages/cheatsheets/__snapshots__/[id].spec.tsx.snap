// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cheatsheet page renders a Cheatsheet page 1`] = `
<DocumentFragment>
  <div
    class="layout__Container-sc-1rx37dj-0 gYqtRt"
  >
    <header
      class="layout__Header-sc-1rx37dj-1 fbezYz"
    >
      <a>
        <img
          alt="The Manurepo"
          class="layout__HeaderImage-sc-1rx37dj-2 dbiPNq"
          src="/images/repo.svg"
        />
      </a>
      <h2
        class="headingLg"
      >
        <a
          class="colorInherit"
          href="/"
        >
          The Manurepo
        </a>
      </h2>
    </header>
    <main>
      <article>
        <div>
          [object Object]
        </div>
      </article>
    </main>
    <div
      class="layout__BackToHomeContainer-sc-1rx37dj-3 dDWKJU"
    >
      <a
        href="/"
      >
        ‚Üê Back to home
      </a>
    </div>
  </div>
</DocumentFragment>
`;
